doc/overlays.xml

```diff
@@ -62,37 +59,40 @@ the overlay, and add a symbolic link to it in the
 <section xml:id="sec-overlays-layout">
 <title>Overlays Layout</title>
 
-<para>An overlay is a Nix expression, which is a function which accepts 2
-arguments.</para>
+<para>Overlays are expressed as Nix functions which accept 2 arguments and return a set of
+packages</para>
 
 <programlisting>
 self: super:
 
 {
-  foo = super.foo.override { ... };
-  bar = import ./pkgs/bar {
-    inherit (self) stdenv fetchurl;
-    inherit (self) ninja crawl dwarf-fortress;
+  boost = super.boost.override {
+    python = self.python3;
+  };
+  rr = super.callPackage ./pkgs/rr {
```

Comment:
Technically you could use both, but this would get some extra meaning in the futur (security-update branch), because `self.callPackage` goes twice through the `self` attribute evaluation.  The first one to resolve the `callPackage` function, and the second time to resolve the dependencies from `self` which are captured by the `callPackage` function.

Note that `self` is the same provided to all layers.  So using `self.callPackage` is for the moment identical to `super.callPackage`, except for this extra hop through the fix-point.

I documented it as functions should be taken from `super`, because they already capture `self` if they have to.

Also note that this way, we make it syntactically verifiable, as anything which comes around the dependencies uses `super`, and anything which is use to find dependencies uses `self`.